### PR TITLE
feat: add blog post for 3.0.0 release

### PR DESCRIPTION
**Pull request by codebytere**
🙅  DO NOT MERGE 🙅 

RFC: please comment with any changes you'd like to see in our preliminary post for the release of `v3.0.0`!

/cc @electron/mergers

### CODE REVIEW

File: data/blog/electron-3-0.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+* [[#13397](https://github.com/electron/electron/pull/13397)] fix: issue with fs.statSyncNoException throwing exceptions
+* [[#13476](https://github.com/electron/electron/pull/13476), [#13452](https://github.com/electron/electron/pull/13452)] fix: crash when loading site with jquery
+* [[#14092](https://github.com/electron/electron/pull/14092)] fix: crash in net::ClientSocketHandle destructor
+* [[#14453](https://github.com/electron/electron/pull/14453)] fix: notify focus change right away rather not on next tick
+* [[#14205](https://github.com/electron/electron/pull/14205)] fix: pressing escape on a dialog now closes it even if defaultId is set
+
+### MacOS
+
+* [[#13220](https://github.com/electron/electron/pull/13220)] fix: issue allowing bundles to be selected in `<input file="type">` open file dialog
```
[the remark this answers is not in the document]
Comment (codebytere) [reply data/blog/electron-3-0.md]: I think we can safely ignore this one ☝️

File: data/blog/electron-3-0.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
+---
+title: Electron 3.0.0
+author: codebytere
+date: '2018-09-20'
+---
+
+The Electron team is excited to annouce that the first stable release of Electron `v3.0.0` is now
+available from [electronjs.org](https://electronjs.org/)! It's jam-packed with upgrades, fixes, and new features, and we can't wait to see what you build with them. Below are more details about the specifics of this latest release, and we welcome your feedback as you explore.
```
Comment (zeke) [inline data/blog/electron-3-0.md]: Might wanna put a one-liner at the top here like `npm i electron@latest --save-dev`

File: data/blog/electron-3-0.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+---
+
+## Release Process
+
+As we undertook development of `v3.0.0`, we sought to more empirically define criteria for a stable release by formalizing the feedback progress for progressive beta releases. This feedback process involved a number of our consumers in Electron's ecosystem, including but not limited to: Slack, VSCode, Atom, OpenFin, Symphony, and Oculus. We wouldn't have been able to mark this milestone without their work to let us know about the blockers and needs they faced with each successive beta as they journeyed down the road to `v3.0.0` upgrades for their apps.
```
Comment (sofianguy) [inline data/blog/electron-3-0.md]: Should we include a statement for those interested in participating in future cycles of the App Feedback Program to email info@electronjs.org?

Comment (sofianguy) [inline data/blog/electron-3-0.md]: add: MS Teams and Atlassian

Comment (codebytere) [reply data/blog/electron-3-0.md]: definitely ☝️

File: data/blog/electron-3-0.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+* [[#12983](https://github.com/electron/electron/pull/12983)] fix: `Menu.setApplicationMenu(null)` crash
+* [[#12507](https://github.com/electron/electron/pull/12507)] fix: `BrowserWindow.focus()` for offscreen windows
+* [[#12192](https://github.com/electron/electron/pull/12192)] fix: use application name for notifications
+* [[#12388](https://github.com/electron/electron/pull/12388)] fix: use the appname as the tray icon's default tooltip
+
+### Windows
+
+# Security-Related Updates
+
+* [SECURITY] fix: webPreferences inheritance issue. CVE-2018-15685
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: This is a little confusing: if I were reading this as a new user, I'd wonder if this meant that 2.0.x was still affected

File: data/blog/electron-3-0.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sofianguy) [reply data/blog/electron-3-0.md]: I'm going to work on updating the AFP info link for posting in electron/electron

File: data/blog/electron-3-0.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+* [[#12093](https://github.com/electron/electron/pull/12093)] feat: workspace notifications support
+* [[#12496](https://github.com/electron/electron/pull/12496)] feat: `tray.setIgnoreDoubleClickEvents(ignore)` to ignore tray double click events. 
+* [[#12281](https://github.com/electron/electron/pull/12281)] feat: mouse forward functionality on macOS
+* [[#12714](https://github.com/electron/electron/pull/12714)] feat: screen lock / unlock events
+* [[#11959](https://github.com/electron/electron/pull/11959)] feat: show file filter name for accessory view of file dialog
+
+### Linux
+
+* [[#12080](https://github.com/electron/electron/pull/12080)] feat: support for CI testing on arm64 hardware
```
[the remark this answers is not in the document]
Comment (codebytere) [reply data/blog/electron-3-0.md]: oh you're v correct

File: data/blog/electron-3-0.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+* [[#12300](https://github.com/electron/electron/pull/12300)] feat: better GTK+ Menu color support
```
[the remark this answers is not in the document]
Comment (codebytere) [reply data/blog/electron-3-0.md]: no i'll remove it!

File: data/blog/electron-3-0.md (hunk 4 above)
Comment (codebytere) [reply data/blog/electron-3-0.md]: hmm i might not include this section tbh

File: data/blog/electron-3-0.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+* [[#14031](https://github.com/electron/electron/pull/14031)] feat: node `fs` wrappers to make `fs.realpathSync.native` and `fs.realpath.native` available
+
+## Breaking API changes
+
+* [[#12362](https://github.com/electron/electron/pull/12362)] feat: updates to menu item order control
+* [[#12991](https://github.com/electron/electron/pull/12991)] refactor: aliased arm builds are no longer published
+* [[#13050](https://github.com/electron/electron/pull/13050)] refactor: removed documented deprecated APIs
+* [[#12477](https://github.com/electron/electron/pull/12477)] refactor: removed `did-get-response-details` and `did-get-redirect-request` events
+* [[#12655](https://github.com/electron/electron/pull/12655)] feat: default to disabling navigating on drag/drop
+* [[#12993](https://github.com/electron/electron/pull/12993)] feat: Node 4.x or greater is required
```
Comment (MarshallOfSound) [inline data/blog/electron-3-0.md]: "... required to use the "electron" npm module"

File: data/blog/electron-3-0.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
+---
+title: Electron 3.0.0
+author: codebytere
+date: '2018-09-20'
+---
+
+The Electron team is excited to annouce that the first stable release of Electron `v3.0.0` is now
```
[the remark this answers is not in the document]
Comment (codebytere) [reply data/blog/electron-3-0.md]: oops should've been `v3.0` 🙈

File: data/blog/electron-3-0.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+* [[#13068](https://github.com/electron/electron/pull/13068)] feat: netLog API for dynamic logging control
+* [[#13539](https://github.com/electron/electron/pull/13539)] feat: enable `webview` in sandbox renderer
+* [[#14118](https://github.com/electron/electron/pull/14118)] feat: `fs.readSync` now works with massive files
+* [[#14031](https://github.com/electron/electron/pull/14031)] feat: node `fs` wrappers to make `fs.realpathSync.native` and `fs.realpath.native` available
+
+## Breaking API changes
+
+* [[#12362](https://github.com/electron/electron/pull/12362)] feat: updates to menu item order control
+* [[#12991](https://github.com/electron/electron/pull/12991)] refactor: aliased arm builds are no longer published
+* [[#13050](https://github.com/electron/electron/pull/13050)] refactor: removed documented deprecated APIs
```
[the remark this answers is not in the document]
Comment (codebytere) [reply data/blog/electron-3-0.md]: oh yeah i think that's most discoverable, i'll make it a bullet list below

File: data/blog/electron-3-0.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+## Changes / New Features
+
+Major bumps to several important parts of Electron's toolchain, including Chrome `v66.0.3359.181`, Node `v10.2.0`, and V8 `v6.6.346.23.`
+
+* [[#12656](https://github.com/electron/electron/pull/12656)] feat: `app.isPackaged`
+* [[#12652](https://github.com/electron/electron/pull/12652)] feat: `app.whenReady()`
+* [[#13183](https://github.com/electron/electron/pull/13183)] feat: `process.getHeapStatistics()`
+* [[#12485](https://github.com/electron/electron/pull/12485)] feat: `win.moveTop()` to move window z-order to top
+* [[#13110](https://github.com/electron/electron/pull/13110)] feat: TextField and Button APIs
```
[the remark this answers is not in the document]
Comment (codebytere) [reply data/blog/electron-3-0.md]: Added in `Other Notes`

File: data/blog/electron-3-0.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (codebytere) [reply data/blog/electron-3-0.md]: Linked to breaking change list ☝️

File: data/blog/electron-3-0.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+---
+
+The Electron team is excited to annouce that the first stable release of Electron `v3.0` is now
+available from [electronjs.org](https://electronjs.org/) and via `npm install electron@latest`! It's jam-packed with upgrades, fixes, and new features, and we can't wait to see what you build with them. Below are details about this release, and we welcome your feedback as you explore.
+
+---
+
+## Release Process
+
+As we undertook development of `v3.0.0`, we sought to more empirically define criteria for a stable release by formalizing the feedback progress for progressive beta releases. This feedback process involved a number of our consumers in Electron's ecosystem, including but not limited to: Slack, VSCode, Atom, OpenFin, Symphony, and Oculus. `v3.0.0` would not have been possible without our [App Feedback Program](LINK-TBD) partners, who provided early testing and feedback during the beta cycle. Thanks to Atlassian, Atom, Microsoft Teams, Oculus, OpenFin, Slack, Symphony, VS Code, and other program members for their work. If you'd like to participate in future betas, please mail us at [info@electronjs.org](mailto:info@electronjs.org).
```
[the remark this answers is not in the document]
Comment (codebytere) [reply data/blog/electron-3-0.md]: oh oops forgot to delete!

File: data/blog/electron-3-0.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
+---
+title: Electron 3.0.0
+author: codebytere
+date: '2018-09-20'
+---
+
+The Electron team is excited to annouce that the first stable release of Electron `v3.0` is now
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: * s/annouce/announce/

 * nit: `v3.0` isn't a version number. Maybe "first stable release of Electron 3"?

File: data/blog/electron-3-0.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+* [[#13110](https://github.com/electron/electron/pull/13110)] feat: TextField and Button APIs
+* [[#13068](https://github.com/electron/electron/pull/13068)] feat: netLog API for dynamic logging control
+* [[#13539](https://github.com/electron/electron/pull/13539)] feat: enable `webview` in sandbox renderer
+* [[#14118](https://github.com/electron/electron/pull/14118)] feat: `fs.readSync` now works with massive files
+* [[#14031](https://github.com/electron/electron/pull/14031)] feat: node `fs` wrappers to make `fs.realpathSync.native` and `fs.realpath.native` available
+
+## Breaking API changes
+
+* [[#12362](https://github.com/electron/electron/pull/12362)] feat: updates to menu item order control
+* [[#12991](https://github.com/electron/electron/pull/12991)] refactor: aliased arm builds are no longer published
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: I get that this is technically a breaking change, but still... this should be moved to the bottom of the list :smile:

File: data/blog/electron-3-0.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+* [[#12477](https://github.com/electron/electron/pull/12477)] refactor: removed `did-get-response-details` and `did-get-redirect-request` events
+* [[#12655](https://github.com/electron/electron/pull/12655)] feat: default to disabling navigating on drag/drop
+* [[#12993](https://github.com/electron/electron/pull/12993)] feat: Node `v4.x` or greater is required use the `electron` npm module
+* [[#12008](https://github.com/electron/electron/pull/12008) [#12140](https://github.com/electron/electron/pull/12140) [#12503](https://github.com/electron/electron/pull/12503) [#12514](https://github.com/electron/electron/pull/12514) [#12584](https://github.com/electron/electron/pull/12584) [#12596](https://github.com/electron/electron/pull/12596) [#12637](https://github.com/electron/electron/pull/12637) [#12660](https://github.com/electron/electron/pull/12660) [#12696](https://github.com/electron/electron/pull/12696) [#12716](https://github.com/electron/electron/pull/12716) [#12750](https://github.com/electron/electron/pull/12750) [#12787](https://github.com/electron/electron/pull/12787) [#12858](https://github.com/electron/electron/pull/12858)] refactor: `NativeWindow`
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: This is kind of awesome

File: data/blog/electron-3-0.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+
+**Nota Bene:** Switching to an older version of Electron after running this version will require you to clear out your user data directory to avoid older versions crashing. You can get the user data directory by running `console.log(app.getPath("userData"))` or see [docs](https://electronjs.org/docs/api/app#appgetpathname) for more details.
+
+## Bug Fixes
+
+* [[#13397](https://github.com/electron/electron/pull/13397)] fix: issue with fs.statSyncNoException throwing exceptions
+* [[#13476](https://github.com/electron/electron/pull/13476), [#13452](https://github.com/electron/electron/pull/13452)] fix: crash when loading site with jquery
+* [[#14092](https://github.com/electron/electron/pull/14092)] fix: crash in net::ClientSocketHandle destructor
+* [[#14453](https://github.com/electron/electron/pull/14453)] fix: notify focus change right away rather not on next tick
+* [[#14205](https://github.com/electron/electron/pull/14205)] fix: pressing escape on a dialog now closes it even if defaultId is set
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: 14205 should be in the MacOS section

File: data/blog/electron-3-0.md [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+
+### MacOS
+
+* [[#13220](https://github.com/electron/electron/pull/13220)] fix: issue allowing bundles to be selected in `<input file="type">` open file dialog
+* [[#12404](https://github.com/electron/electron/pull/12404)] fix: issue blocking main process when using async dialog
+* [[#12395](https://github.com/electron/electron/pull/12395)] fix: dark mode tray highlighting bug
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: 12395 is already in 2.0.x, so shouldn't be in this announcement as new in 3.0.0

File: data/blog/electron-3-0.md [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+* [[#12043](https://github.com/electron/electron/pull/12043)] fix: context menu click callback
+* [[#12527](https://github.com/electron/electron/pull/12527)] fix: event leak on reuse of touchbar item
+* [[#12352](https://github.com/electron/electron/pull/12352)] fix: tray title crash
+* [[#12327](https://github.com/electron/electron/pull/12327)] fix: non-draggable regions
+* [[#12809](https://github.com/electron/electron/pull/12809)] fix: to prevent menu update while it's open
+* [[#13162](https://github.com/electron/electron/pull/13162)] fix: tray icon bounds not allowing negative values
+* [[#13085](https://github.com/electron/electron/pull/13085)] fix: tray title not inverting when highlighted
+* [[#12196](https://github.com/electron/electron/pull/12196)] fix: Mac build when `enable_run_as_node==false`
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: 12196 is about a FTBFS fix, isn't a user-visible change

File: data/blog/electron-3-0.md [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+* [[#12157](https://github.com/electron/electron/pull/12157)] fix: additional issues on frameless windows with vibrancy
+* [[#13326](https://github.com/electron/electron/pull/13326)] fix: to set mac protocol to none after calling `app.removeAsDefaultProtocolClient`
+* [[#13530](https://github.com/electron/electron/pull/13530)] fix: incorrect usage of private APIs in MAS build
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: 13530 is already in 2.0.x, so shouldn't be in this announcement as new in 3.0.0

File: data/blog/electron-3-0.md [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+* [[#13517](https://github.com/electron/electron/pull/13517)] fix: `tray.setContextMen`u crash
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: 13517 is already in 2.0.x, so shouldn't be in this announcement as new in 3.0.0

File: data/blog/electron-3-0.md [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+* [[#13085](https://github.com/electron/electron/pull/13085)] fix: tray title not inverting when highlighted
+* [[#12196](https://github.com/electron/electron/pull/12196)] fix: Mac build when `enable_run_as_node==false`
+* [[#12157](https://github.com/electron/electron/pull/12157)] fix: additional issues on frameless windows with vibrancy
+* [[#13326](https://github.com/electron/electron/pull/13326)] fix: to set mac protocol to none after calling `app.removeAsDefaultProtocolClient`
+* [[#13530](https://github.com/electron/electron/pull/13530)] fix: incorrect usage of private APIs in MAS build
+* [[#13517](https://github.com/electron/electron/pull/13517)] fix: `tray.setContextMen`u crash 
+
+### Linux
+
+* [[#12983](https://github.com/electron/electron/pull/12983)] fix: `Menu.setApplicationMenu(null)` crash
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: 12983 is already in 2.0.x, so shouldn't be in this announcement as new in 3.0.0

File: data/blog/electron-3-0.md [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+* [[#12507](https://github.com/electron/electron/pull/12507)] fix: `BrowserWindow.focus()` for offscreen windows
+* [[#12192](https://github.com/electron/electron/pull/12192)] fix: use application name for notifications
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: 12192 is already in 2.0.x, so shouldn't be in this announcement as new in 3.0.0

File: data/blog/electron-3-0.md [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+* [[#12388](https://github.com/electron/electron/pull/12388)] fix: use the appname as the tray icon's default tooltip
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: 12388 is already in 2.0.x, so shouldn't be in this announcement as new in 3.0.0

File: data/blog/electron-3-0.md [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+
+## Other Notes
+
+* PDF Viewer is currently not working but is being worked on and will be functional once again soon
+* `TextField` and `Button` APIs are experimental and are therefore off by default
+  * They can be enabled with the `enable_view_api` build flag
+
+# What's Next
+
+The Electron team continues to work on defining our processes for more rapid and smooth upgrades as we seek to ultimately maintain parity with the latest versions of Chromium, Node, and V8.
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: maybe s/parity/cadence/ ?

File: data/blog/electron-3-0.md [hunk 24]
```diff
@@ -0,0 +1,101 @@
...
+---
+
+The Electron team is excited to annouce that the first stable release of Electron `v3.0` is now
+available from [electronjs.org](https://electronjs.org/) and via `npm install electron@latest`! It's jam-packed with upgrades, fixes, and new features, and we can't wait to see what you build with them. Below are details about this release, and we welcome your feedback as you explore.
+
+---
+
+## Release Process
+
+As we undertook development of `v3.0.0`, we sought to more empirically define criteria for a stable release by formalizing the feedback progress for progressive beta releases. `v3.0.0` would not have been possible without our [App Feedback Program](LINK-TBD) partners, who provided early testing and feedback during the beta cycle. Thanks to Atlassian, Atom, Microsoft Teams, Oculus, OpenFin, Slack, Symphony, VS Code, and other program members for their work. If you'd like to participate in future betas, please mail us at [info@electronjs.org](mailto:info@electronjs.org).
```
Comment (ckerr) [inline data/blog/electron-3-0.md]: LINK-TBD should be

https://electronjs.org/docs/tutorial/app-feedback-program 

(I think?)

File: data/blog/electron-3-0.md (hunk 13 above)
Comment (codebytere) [reply data/blog/electron-3-0.md]: i think i'll remove all chore-type PRs, they're not user facing so i doubt anyone will miss them 😆

File: data/blog/electron-3-0.md (hunk 14 above)
Comment (codebytere) [reply data/blog/electron-3-0.md]: 💃

**Pull request merged**